Lib/test/test_patma.py

```diff
@@ -2654,6 +2654,20 @@ def get(key, default=None):
 
         self.assertEqual(y, 'bar')
 
+    def test_patma_249(self):
+        class C:
+            __attr = "_C__attr"
+        class Outer:
+            __attr = "_Outer__attr"
+            def f(self, x):
+                match x:
+                    # looks up __attr, not _C__attr or _Outer__attr
+                    case C(__attr=y):
+                        return y
+        c = C()
+        setattr(c, "__attr", "__attr")  # setattr is needed because we're in a class scope
+        self.assertEqual(Outer().f(c), "__attr")
```

Comment:
I'd use something other than `"__attr"` for the *value* here, to avoid being too cute.
```suggestion
        setattr(c, "__attr", "spam")  # setattr is needed because we're in a class scope
        self.assertEqual(Outer().f(c), "spam")
```

Reply:
Done